plugin/trino-druid/src/test/java/io/trino/plugin/druid/DruidCreateAndInsertDataSetup.java

```diff
@@ -0,0 +1,109 @@
+import io.trino.plugin.druid.ingestion.TimestampSpec;
+import io.trino.testing.datatype.ColumnSetup;
+import io.trino.testing.datatype.DataSetup;
+import io.trino.testing.sql.TemporaryRelation;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class DruidCreateAndInsertDataSetup
+        implements DataSetup
+{
+    private static final Logger log = Logger.get(DruidCreateAndInsertDataSetup.class);
+    private final TestingDruidServer druidServer;
+    private final String dataSourceNamePrefix;
+
+    public DruidCreateAndInsertDataSetup(TestingDruidServer druidServer, String dataSourceNamePrefix)
+    {
+        this.druidServer = druidServer;
+        this.dataSourceNamePrefix = dataSourceNamePrefix;
+    }
+
+    @Override
+    public TemporaryRelation setupTemporaryRelation(List<ColumnSetup> inputs)
+    {
+        DruidTable testTable = new DruidTable(this.dataSourceNamePrefix);
+        try {
+            ingestData(testTable, inputs);
+        }
+        catch (Exception e) {
+            log.error(e);
+        }
+        return testTable;
+    }
+
+    private void ingestData(DruidTable testTable, List<ColumnSetup> inputs)
+            throws Exception
+    {
+        IndexTaskBuilder builder = new IndexTaskBuilder();
+        builder.setDatasource(testTable.getName());
+        TimestampSpec timestampSpec = getTimestampSpec(inputs);
+        builder.setTimestampSpec(timestampSpec);
+
+        List<ColumnSetup> normalInputs = inputs.stream().filter(input -> !isTimestampDimension(input)).collect(Collectors.toList());
+        for (int index = 0; index < inputs.size() - 1; index++) {
+            builder.addColumn(format("col_%s", index), normalInputs.get(index).getDeclaredType().orElse("string"));
+        }
+
+        String dataFilePath = format("%s/%s.tsv", druidServer.getHostWorkingDirectory(), testTable.getName());
+        writeTsvFile(dataFilePath, inputs);
+
+        log.debug(builder.build());
+        this.druidServer.ingestData(testTable.getName(), builder.build(), dataFilePath);
+    }
+
+    private TimestampSpec getTimestampSpec(List<ColumnSetup> inputs)
+    {
+        List<ColumnSetup> timestampInputs = inputs.stream().filter(this::isTimestampDimension).collect(Collectors.toList());
+
+        if (timestampInputs.size() > 1) {
+            throw new UnsupportedOperationException("Druid only allows one timestamp field");
+        }
+
+        return new TimestampSpec("dummy_druid_ts", "auto");
+    }
+
+    private boolean isTimestampDimension(ColumnSetup input)
+    {
+        if (input.getDeclaredType().isEmpty()) {
+            return false;
+        }
+        String type = input.getDeclaredType().get();
+
+        // TODO: support more types
```

Comment:
I don't get this comment. Do you mean more timestamp types?

Reply:
That comment is invalid. Have removed it